### ISSUE TITLE
support $eval()ed handlers

### ISSUE THREAD

**Issue by zacronos** [realtymaps/rmaps-angular-utils#1]
The ng-replacements code is currently a little awkward --  it requires 2 attributes where the native angular versions (e.g. `ng-click`) do not.

The implementation could be made much more flexible and intuitive if, instead of having 2 separate attributes for handler function and value, there were a single attribute which just gets passed to `scope.$eval()`.  It would allow things like `rmaps-click="handler(value)"` or even `rmaps-click="handler(obj.id+'_suffix')", as is common in angular.

This issue is created for discussion of whether such a change is a good idea.

@nmccready @jessecstewart

**Comment by nmccready** [comment]
I disagree, if you start using $eval your not doing anything different then what ng-click did originally. The angular source code base of $eval calls $parse. $parse is what ng-click normally calls. This adds overhead and watches.

If you need another set of ngReplacements (some other name), then I think there should be another set . Not replacing this one as it is supposed to be a bare minimum of no angular magic.

Supporting Lines:
https://github.com/angular/angular.js/blob/master/src/ng/directive/ngEventDirs.js#L60

https://github.com/angular/angular.js/blob/291d7c467fba51a9cb89cbeee62202d51fe64b09/src/ng/rootScope.js#L943-L945

**Comment by zacronos** [comment]
Are you sure $parse sets up watches?  I only made this suggestion because I thought we had a different flow that avoids watches, but you are correct that the only change seems to be the use of $quickParse instead of $parse.  Unless we are certain that $parse uses watches for some reason, we should change the documentation to remove all references to eliminating watches.

Did you run performance tests with $parse is used with `expensiveChecks` set to false instead of true?  I'm curious how much of a difference that makes.  https://github.com/angular/angular.js/blob/master/src/ng/directive/ngEventDirs.js#L60

**Comment by nmccready** [comment]
I never noticed expensiveChecks as that might be new. I didn't do "performance" tests where I was timing stuff per-say. However our test was results list being crap slow vs fast with the way it is now.  So should this be pursued further and explored?

**Comment by nmccready** [comment]
Also notice that ng-"event" never unbinds an event. So that has to be a leak.

**Comment by nmccready** [comment]
@zacronos just messed with the code and it appears that ng-click ng-mouseover are much faster now. So this directive set may not be needed any longer. But if you want we can keep it around and set expensive checks to false.
